### PR TITLE
Add test pipeline parameters

### PR DESCRIPTION
**Pull request by llxia**
- Allow users to set ADOPTOPENJDK_REPO, ADOPTOPENJDK_BRANCH, USE_TESTENV_PROPERTIES and EXTRA_OPTIONS in Pipeline-Build-Test-Personal job
- If ADOPTOPENJDK_REPO is set to personal repo, auto-set GENERATE_JOBS = true and LIGHT_WEIGHT_CHECKOUT = false

resolves: https://github.com/eclipse-openj9/openj9/issues/19231

### CODE REVIEW

**Comment by llxia** [comment]
1. [link](https://hyc-runtimes-jenkins.swg-devops.com/job/Pipeline-Build-Test-Personal/21583/)
   - ADOPTOPENJDK_REPO = https://github.com/llxia/openjdk-tests.git
   - ADOPTOPENJDK_BRANCH = test2
   - USE_TESTENV_PROPERTIES = true
   - EXTRA_OPTIONS = -Xjit
   - [test job](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk11_j9_sanity.perf_aarch64_linux_Personal/17)
2. [link](https://hyc-runtimes-jenkins.swg-devops.com/job/Pipeline-Build-Test-Personal/21585/)
   - ADOPTOPENJDK_REPO = https://github.com/adoptium/aqa-tests.git
   - ADOPTOPENJDK_BRANCH = master
   - USE_TESTENV_PROPERTIES = true
   - EXTRA_OPTIONS = -Xjit
   - [test job](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk11_j9_sanity.perf_aarch64_linux_Personal/18)

**Comment by llxia** [comment]
jenkins test sanity xlinux jdk11

**Comment by llxia** [comment]
jenkins test sanity xlinux jdk11

**Review by pshipton — APPROVED** [review]

File: buildenv/jenkins/common/variables-functions.groovy [hunk 1]
```diff
@@ -1058,6 +1058,30 @@ def set_job_variables(job_type) {
     USE_TESTENV_PROPERTIES = params.USE_TESTENV_PROPERTIES ?: false
     echo "Using USE_TESTENV_PROPERTIES = ${USE_TESTENV_PROPERTIES}"
 
+    if (PERSONAL_BUILD && PERSONAL_BUILD.equalsIgnoreCase('true')) {
+        LIGHT_WEIGHT_CHECKOUT = true
+        GENERATE_JOBS = params.AUTOMATIC_GENERATION ?: false
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/variables-functions.groovy:1063]: I think we need to always generate. We don't know what ran last. If user has default params and previous run was a personal repo, we need to put it back to default.

Comment (llxia) [reply buildenv/jenkins/common/variables-functions.groovy:1063]: AUTOMATIC_GENERATION is true by default. It means GENERATE_JOBS is true (always generate the test job).

Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions.groovy:1063]: Ah good point. So, shouldn't the code be `GENERATE_JOBS = params.AUTOMATIC_GENERATION ?: true` then? Why would we default to false if it is not defined?

Comment (llxia) [reply buildenv/jenkins/common/variables-functions.groovy]: updated

**Comment by llxia** [comment]
@AdamBrousseau could you review the updated PR? Thanks

**Review by AdamBrousseau — APPROVED** [review]

**Pull request merged**